openmc/tally_derivative.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import division
+
+import sys
+from numbers import Integral
+from xml.etree import ElementTree as ET
+
+from six import string_types
+
+import openmc.checkvalue as cv
+
+
+# "Static" variable for auto-generated TallyDerivative IDs
+AUTO_TALLY_DERIV_ID = 10000
```

Comment:
We don't start at 1 with any of the other auto-IDs.  I suspect the logic behind this is that a person might want to manually specify the ID of some derivatives but not others.  So you'd want to be able to say
```python
boring_deriv = openmc.TallyDerivative()
important_deriv = openmc.TallyDerivative(derivative_id=1)
```